forc-plugins/forc-client/src/op/submit.rs

```diff
@@ -0,0 +1,96 @@
+use crate::cmd;
+use anyhow::Context;
+use fuel_gql_client::{
+    client::{types::TransactionStatus, FuelClient},
+    fuel_tx,
+};
+
+/// A command for submitting transactions to a Fuel network.
+pub async fn submit(cmd: cmd::Submit) -> anyhow::Result<()> {
+    let tx = read_tx(&cmd.tx_path)?;
+    let client = FuelClient::new(&cmd.network.node_url)?;
+    if cmd.network.await_ {
+        let status = client
+            .submit_and_await_commit(&tx)
+            .await
+            .context("Submission of tx or awaiting commit failed")?;
+        if cmd.tx_status.json {
+            print_status_json(&status)?;
+        } else {
+            print_status(&status);
+        }
+    } else {
+        let id = client.submit(&tx).await.context("Failed to submit tx")?;
+        println!("{id}");
+    }
+    Ok(())
+}
+
+/// Deserialize a `Transaction` from the given file into memory.
+pub fn read_tx(path: &std::path::Path) -> anyhow::Result<fuel_tx::Transaction> {
+    let file = std::fs::File::open(path)?;
+    let reader = std::io::BufReader::new(file);
+    let tx: fuel_tx::Transaction = if path.ends_with("json") {
```

Comment:
While testing,  I realized that even if the file has `json` or `bin` extension it fails with invalid file format error so, maybe we should be using `.extension()` for checking the extension ([relevant std reference](https://doc.rust-lang.org/std/path/struct.Path.html#method.ends_with)).

Reply:
Ahh good catch!

In a future PR I'd like to add a section to our E2E testing that tests these CLI commands, but want to try making it a little easier to construct transaction files to test with first.